pkg/controller/common/settings/replace.go

```diff
@@ -0,0 +1,35 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package settings
+
+// Replacement is a utility to replace an element in an untypedDict at Path with Replacement iff the element is equal
+// to Expected. This is mainly intended to enable a workaround around for https://github.com/elastic/cloud-on-k8s/issues/3718
+type Replacement struct {
+	Path        []string
+	Expected    interface{}
+	Replacement interface{}
+}
+
+func (r Replacement) apply(out untypedDict) {
+	replaceTarget := out
+	for i, p := range r.Path {
+		// last element?
+		if i == len(r.Path)-1 {
+			// matches expected value?
+			actual, exists := replaceTarget[p]
+			if exists && actual == r.Expected {
+				// do the replacement
+				replaceTarget[p] = r.Replacement
+				return
+			}
+		}
+
+		v, exists := replaceTarget[p].(untypedDict)
+		if !exists {
+			return
+		}
+		replaceTarget = v
```

Comment:
The 4 lines above give me a headache for some reason 😄 
we mutate `replaceTarget` by the inner path value here, right? A comment would help, but also variable names a bit more explicit than `v` and `p`.

Reply:
You are right, it is too terse. I revised the naming.